iron-dropdown-scroll-manager.html

```diff
@@ -33,39 +36,45 @@
 
 
       /**
-       * Returns true if the provided element is "scroll locked," which is to
-       * say that it cannot be scrolled via pointer or keyboard interactions.
+       * Returns true if the element causing the scroll is "scroll locked," which is to
+       * say that it cannot be scrolled via pointer or keyboard interactions, or
+       * if already at its scroll boundaries.
        *
-       * @param {HTMLElement} element An HTML element instance which may or may
-       * not be scroll locked.
+       * @param {Event} event The scroll event
        */
-      elementIsScrollLocked: function(element) {
-        var currentLockingElement = this.currentLockingElement;
-
-        if (currentLockingElement === undefined)
+      shouldPreventScrolling: function(event) {
+        // Avoid expensive checks if the event is not one of the observed keys.
+        if (event.type === 'keydown' && !this._isScrollingKeypress(event)) {
           return false;
-
-        var scrollLocked;
-
-        if (this._hasCachedLockedElement(element)) {
-          return true;
         }
 
-        if (this._hasCachedUnlockedElement(element)) {
+        var currentLockingElement = this.currentLockingElement;
+        if (currentLockingElement === undefined)
           return false;
-        }
 
-        scrollLocked = !!currentLockingElement &&
-          currentLockingElement !== element &&
-          !this._composedTreeContains(currentLockingElement, element);
+        var element = /** @type {HTMLElement} */(Polymer.dom(event).rootTarget);
+        var scrollables = this._scrollableParentNodes(element, currentLockingElement);
 
-        if (scrollLocked) {
-          this._lockedElementCache.push(element);
-        } else {
-          this._unlockedElementCache.push(element);
+        var isScrollingRight = event.wheelDeltaX > 0;
+        var isScrollingUp = event.wheelDeltaY > 0;
+        if (event.type === 'touchmove' && lastTouch) {
+          isScrollingUp = event.pageY > lastTouch.pageY;
+          isScrollingRight = lastTouch.pageX > event.pageX;
         }
 
-        return scrollLocked;
+        // Search for one scrollable that can still scroll.
+        // If none, prevent event from bubbling up to the window.
+        var shouldPrevent = true;
+        for (var i = 0; i < scrollables.length && shouldPrevent; i++) {
+          var scrollable = scrollables[i];
+          var canScrollVertically = isScrollingUp ? scrollable.scrollTop > 0 :
+              scrollable.scrollTop + scrollable.offsetHeight < scrollable.scrollHeight;
```

Comment:
I believe this should be `scrollable.clientHeight` and `scrollable.clientWidth`  I've making this mistake too :(

> 
clientHeight:
The Element.clientHeight read-only property is zero for elements with no CSS or inline layout boxes, otherwise it's  the inner height of an element in pixels, including padding but not the horizontal scrollbar height, border, or margin.